src/type_checker.cpp

```diff
@@ -190,6 +186,10 @@ void TypeChecker::Visit(FuncDefNode& func_def) {
     // TODO: redefinition of function id
   }
 
+  if (func_def.id == "main") {
+    has_main_func_ = true;
+  }
```

Comment:
I don't recall why we were checking for the existence of the main function, but I believe this check is unnecessary now. It's perfectly acceptable for a translation unit to not have a main function. The linker will handle this and report any issues.